config/bidderinfo.go

```diff
@@ -260,7 +254,25 @@ func processBidderInfos(reader InfoReader, normalizeBidderName func(string) (ope
 					return nil, fmt.Errorf("error parsing config for aliased bidder %s: %v", fileName, err)
 				}
 
+				//required for CoreBidderNames function to also return aliasBiddernames
+				if err := openrtb_ext.SetAliasBidderName(bidderName[0], openrtb_ext.BidderName(info.AliasOf)); err != nil {
+					return nil, err
+				}
+
+				normalizedBidderName, bidderNameExists := normalizeBidderName(bidderName[0])
+				if !bidderNameExists {
+					return nil, fmt.Errorf("error parsing config for an alias %s: unknown bidder", fileName)
+				}
+
 				aliasNillableFieldsByBidder[string(normalizedBidderName)] = aliasFields
+				bidderInfos[string(normalizedBidderName)] = info
+			} else {
+				normalizedBidderName, bidderNameExists := normalizeBidderName(bidderName[0])
+				if !bidderNameExists {
+					return nil, fmt.Errorf("error parsing config for bidder %s: unknown bidder", fileName)
+				}
+
+				bidderInfos[string(normalizedBidderName)] = info
```

Comment:
This code can be refactored to avoid duplication and else case is not needed: 
```go
if len(bidderName) == 2 && bidderName[1] == "yaml" {
			info := BidderInfo{}
			if err := yaml.Unmarshal(data, &info); err != nil {
				return nil, fmt.Errorf("error parsing config for bidder %s: %v", fileName, err)
			}

			normalizedBidderName, bidderNameExists := normalizeBidderName(bidderName[0])
			if !bidderNameExists {
				return nil, fmt.Errorf("error parsing config for an alias %s: unknown bidder", fileName)
			}

			//need to maintain nullable fields from BidderInfo struct into bidderInfoNullableFields
			//to handle the default values in aliases yaml
			if len(info.AliasOf) > 0 {
				aliasFields := aliasNillableFields{}
				if err := yaml.Unmarshal(data, &aliasFields); err != nil {
					return nil, fmt.Errorf("error parsing config for aliased bidder %s: %v", fileName, err)
				}

				//required for CoreBidderNames function to also return aliasBiddernames
				if err := openrtb_ext.SetAliasBidderName(bidderName[0], openrtb_ext.BidderName(info.AliasOf)); err != nil {
					return nil, err
				}
				aliasNillableFieldsByBidder[string(normalizedBidderName)] = aliasFields
			}
			bidderInfos[string(normalizedBidderName)] = info
		}
```

Reply:
Unfortunately, we cannot. The reason is `normalizeBidderName` function call loops over CoreBidderNames. However, to get an alias, we need to first add alias to CoreBidderNames before we get `normalizeBidderName` to return an alias name. 
The way that you have suggested the code will error out while calling `normalizeBidderName` for an alias since alias name is being added later while calling `openrtb_ext.SetAliasBidderName`.

Reply:
Oh wait... Do you want to say that aliases will not be added to the `coreBidderNames` list? 
In this case I didn't understand how the feature will work and my local set up was incorrect. 
Yes, I apologize, then this function should stay in a way it is and iterate over {bidder}.yaml files.

Reply:
closing the conversation here - Aliases will be added to coreBidderNames using `openrtb_ext.SetAliasBidderName`